src/main/java/net/canfar/storage/web/resources/GroupNameServerResource.java

```diff
@@ -120,13 +122,19 @@ public void write(final JSONWriter writer) throws JSONException {
                     writer.endArray();
                 } catch (PrivilegedActionException e) {
                     throw new JSONException(e.getException());
+                } catch (Exception e) {
+                    // This will likely come from the token store, which the user would not have control over.
+                    throw new RuntimeException(e.getMessage(), e);
                 }
             }
         };
     }
 
-    final List<String> queryGroupNames(final Subject voSpaceUser) throws PrivilegedActionException {
-        return Subject.doAs(voSpaceUser, (PrivilegedExceptionAction<List<String>>) () -> {
+    final List<String> queryGroupNames() throws Exception {
+        final RegistryClient registryClient = getRegistryClient();
+        final URL gmsServiceURL = registryClient.getServiceURL(storageConfiguration.getGMSServiceURI(), Standards.GMS_SEARCH_10, AuthMethod.TOKEN);
```

Comment:
This a very naive question - is there only ever going to be one `RegistryClient`, and one `gmsServiceURL`? If not, where is multiplicity going to be handled?

Reply:
Correct, only one for now.  When we implement multiple GMSes, we will need to ask the Node where to go for its Group information, but we don't have an API for that yet.

Reply:
Comment added to other method.  Thanks!